daemons/fenced/fenced_commands.c

```diff
@@ -501,7 +516,7 @@ stonith_device_execute(stonith_device_t * device)
                                            cmd->done_cb, fork_cb);
     if (exec_rc < 0) {
         cmd->activating_on = NULL;
-        report_internal_result(cmd, exec_rc);
+        cmd->done_cb(0, stonith__action_result(action), cmd);
```

Comment:
sigh ... with C++ name-mangling the decision would be simple.
going for a function here that takes the result to report it is questionable just for similar encapsulation.

Reply:
Right, I decided not to, since this was the only place that would need it